rewrite-core/src/test/kotlin/org/openrewrite/RecipeSchedulerTest.kt

```diff
@@ -40,7 +38,19 @@ class BoomRecipe : Recipe() {
     override fun getDisplayName() = "We go boom"
     override fun getVisitor() = object : PlainTextVisitor<ExecutionContext>() {
         override fun visitText(text: PlainText, p: ExecutionContext): PlainText {
-            throw IllegalStateException("boom")
+            throw BoomException()
         }
     }
 }
+
+/**
+ * Simplified exception that only displays stack trace elements within the [BoomRecipe].
+ */
+class BoomException: RuntimeException("boom") {
+    override fun getStackTrace(): Array<StackTraceElement> {
+        val stackTrace = super.getStackTrace()
+        return stackTrace
+            .filter { it.className.startsWith(BoomRecipe::class.java.name) }
+            .toTypedArray()
+    }
```

Comment:
Now this test won't break any time anything else in the stack has it's line numbers or internal infrastructure changed.